src/dags/importscripts/import_deelmobiliteit.py

```diff
@@ -0,0 +1,394 @@
+                        which columns uniquely identifiy a record
+        digest_size: size to set the max bytes to use for the hash
+
+    Returns:
+        a hased value of the composite values in int
+
+    Note:
+        The default digest size is for now set to 5 bytes, which is equivalent
+        to ~ 10 karakters long.
+        Because the id column is preferably of type int, the hased value is converted
+        from hex to int.
+    """
+    return int.from_bytes(
+        blake2s("|".join(composite_values).encode(), digest_size=digest_size).digest(),
+        byteorder="little",
+    )
+
+
+# ----------------------------------------------------- #
+# SCOOTER                                               #
+# ----------------------------------------------------- #
+@dataclass
+class Scooter:
+    id: int
+    scooter_id: str
+    datumtijd_ontvangen: datetime
+    geometrie: Point
+    exploitant: str
+    status_motor: bool
+    status_beschikbaar: Optional[str] = None
+    naam: Optional[str] = None
+    max_snelheid: Optional[int] = None
+    huidige_locatie: Optional[str] = None
+    kenteken: Optional[str] = None
+    helm_verplicht: Optional[bool] = None
+    helm_aantal_aanwezig: Optional[int] = None
+    tarief_start: Optional[int] = None
+    tarief_gebruik: Optional[int] = None
+    tarief_parkeren: Optional[int] = None
+
+
+def get_data_scooter_fleyx(api_endpoint: str, api_header: Dict) -> Iterator[Scooter]:
+    """Retreives the data from resource FLEYX
+    Because each resource has it's own quirks when calling the API
+    each resource has it's own get_data_* method.
+
+    Args:
+        api_endpoint: an URL specification for the data service to call
+        api_header: a dictionary of headers key value pairs used when calling
+            the data endpoint
+
+    Yields:
+        A list of Scooter type instances containing FLEYX data
+
+    """
+    try:
+        request = requests.get(api_endpoint, headers=api_header, verify=True)
+    except requests.exceptions.RequestException as e:
+        raise requests.exceptions.RequestException(
+            f"{api_endpoint} something went wrong: {request.status_code}", e
+        ) from e
+    else:
+        data = json.loads(request.text)["data"]["bikes"]
+        rows = []
+        for row in data:
+            current_time = (
+                datetime.now(timezone.utc).astimezone(to_zone).strftime("%Y-%m-%d %H:%M:%S")
+            )
+            scooter_object = Scooter(
+                id=make_hash([row["bike_id"], str(current_time)]),
+                scooter_id=row["bike_id"],
+                datumtijd_ontvangen=datetime.strptime(current_time, "%Y-%m-%d %H:%M:%S"),
+                geometrie=Point(row["lon"], row["lat"]),
+                status_motor=row["is_disabled"],
+                status_beschikbaar=row["is_reserved"],
+                exploitant="fleyx",
+            )
+            rows.append(scooter_object)
```

Comment:
Hier kun je toch ook direct het `scooter_object` yielde, ipv eerst de lijst op te bouwen in `rows`, of mis ik iets?

Reply:
Ja gelijk, aangepast